base/dict.jl

```diff
@@ -400,6 +400,10 @@ SerializationState(io::IO) = SerializationState{typeof(io)}(io)
 
 # dict
 
+# These can be changed, to trade off better performance for space
+const global maxallowedprobe = [16]
+const global maxprobeshift   = [6]
```

Comment:
Agree. If anything, setting these on a per-dict choice makes more sense. Having global parameters like this means that to solve one particular performance problem in one part of the system, you end up changing the behavior of every dict everywhere in the system.